core/iwasm/interpreter/wasm_loader.c

```diff
@@ -4323,7 +4419,17 @@ wasm_loader_unload(WASMModule *module)
 
 #if WASM_ENABLE_FAST_JIT != 0 && WASM_ENABLE_JIT != 0 \
     && WASM_ENABLE_LAZY_JIT != 0
-    module->orcjit_stop_compiling = true;
+#if WASM_ENABLE_DYNAMIC_PGO != 0
+    if (module->orcjit_stop_lock_inited) {
+        os_mutex_lock(&module->orcjit_stop_lock);
+#endif
+        module->orcjit_stop_compiling = true;
```

Comment:
Should also copy this line out of `#if WASM_ENABLE_DYNAMIC_PGO != 0 ... #endif`? Or it will change the original logic when dynamic pgo isn't enabled. And not sure why not `module->orcjit_stop_compiling = true;` before `if (module->orcjit_stop_lock_inited) { `?